lib/signaling/v2/iceconnectionmonitor.js

```diff
@@ -65,18 +87,20 @@ class IceConnectionMonitor {
    * @param {function} onIceConnectionInactive
    */
   start(onIceConnectionInactive) {
-    this.stop();
+    this.clear();
     this._timer = setInterval(() => {
       this._getIceConnectionStats().then(iceStats => {
         if (!iceStats) {
           return;
         }
 
+        // console.log('makarand: iceStats.bytesReceived:', iceStats.bytesReceived);
         if (!this._lastActivity || this._lastActivity.bytesReceived !== iceStats.bytesReceived) {
           this._lastActivity = iceStats;
         }
 
         if (iceStats.timestamp - this._lastActivity.timestamp >= this._inactivityThresholdMs) {
+          this._wasInactive = true;
```

Comment:
@makarandp0 Instead of having a property and a callback here, why don't we just schedule the callback to be called when `iceConnectionState` is `disconnected`? Then, we likely don't have to change anything in PeerConnectionV2:

```js
if (iceStats.timestamp - this._lastActivity.timestamp >= this._inactivityThresholdMs) {
  if (this._peerConnection.iceConnectionState === 'disconnected') {
    onIceConnectionInactive();
  } else {
    this._peerConnection.addEventListener('iceconnectionstatechange', function onIceConnectionStateChanged() {
      if (this._peerConnection.iceConnectionState === 'disconnected') {
        this._peerConnection.removeEventListener('iceconnectionstatechange', onIceConnectionStateChanged);
        onIceConnectionInactive();
      }
    });
  }
}
```

Reply:
thanks @manjeshbhargav I thought about it as well. Your suggestion does look very elegant, but the reason I did not go for it was:

1. Right now all the changes in response to peerConnection.iceconnectionstatechange are handled in single place, making it bit easier to follow the code. This handling would make it bit harder to follow.

2. when the  `onIceConnectionInactive` would fire on PeerConnection.js - it's possible that its own handler for iceconnectionstatechange is not fired yet. Thus it may have its state  'thinking" iceConnection is 'connected`.

Reply:
Hey @manjeshbhargav - It does look much elegant, so I have updated my PR to use your suggestion.